### PR TITLE
More label interpretations

### PR DESCRIPTION
**Pull request by PikalaxALT**
Initial request: HDMA transfer functions, more battle animations, more decoding of messy mobile code, and finish naming every function in main.asm

### CODE REVIEW

File: engine/predef.asm [hunk 1]
```diff
@@ -102,7 +102,7 @@ PredefPointers:: ; 856b
 	add_predef PlaceStatusString
 	add_predef LoadMonAnimation
 	add_predef AnimateFrontpic
-	add_predef Functiond0669 ; $48
+	add_predef Predef48 ; $48
 	add_predef HOF_AnimateFrontpic
-	dbw $ff, Function2d43 ; ????
+	dwb $43ff, $2d
```
Comment (yenatch) [inline engine/predef.asm]: for what purpose

Comment (PikalaxALT) [inline engine/predef.asm]: it doesn't actually reference Function2d43; maybe it's a broken pointer?

Comment (yenatch) [inline engine/predef.asm]: Gold has the same pointer and function (2e16 in that).

```
00008c20  3e 81 79 3e 86 57 14 66  4a 70 ff 16 2e cd 11 53  |>.y>.W.fJp.....S|
```
```
00002e10  cd cd 0d c1 e1 c9 00 00  00 00 00 00 00 00 00 00  |................|
00002e20  00 00 00 00 00 00 c9 ea  56 d1 f0 9f f5 fa 56 d1  |........V.....V.|
```

Comment (PikalaxALT) [inline engine/predef.asm]: Huh.  Weird.  Alright.

File: macros/movement.asm [hunk 2]
```diff
@@ -311,51 +311,16 @@ hide_person: macro
...
-	db \1 ; duration
-	endm
+if \1 <= 8
+	db movement_step_sleep + \1 - 1
+else
+	db movement_step_sleep + 8, \1
+endc
+endm
+
+__enum__ SET $47
```
Comment (yenatch) [inline macros/movement.asm]: __enum__ = __enum__ + 9

File: macros/movement.asm [hunk 3]
```diff
@@ -1,285 +1,104 @@
 
 	enum_start
 
-	enum movement_turn_head_down
-turn_head_down: macro
-	db movement_turn_head_down ; $00
+	enum movement_turn_head
+turn_head: macro
+	db movement_turn_head | \1 ; $00
```
Comment (yenatch) [inline macros/movement.asm]: i think `+` makes more sense here

File: macros/movement.asm [hunk 4]
```diff
@@ -1,285 +1,104 @@
...
-	enum movement_slow_step_up
-slow_step_up: macro
-	db movement_slow_step_up ; $09
-	endm
-
-	enum movement_slow_step_left
-slow_step_left: macro
-	db movement_slow_step_left ; $0a
-	endm
+__enum__ SET $4
```
Comment (yenatch) [inline macros/movement.asm]: and wow yeah get rid of this

File: constants/sprite_constants.asm [hunk 5]
```diff
@@ -517,3 +517,146 @@ MAX_OUTDOOR_SPRITES EQU 23
 	const SPRITE_ANIM_INDEX_2A
 	const SPRITE_ANIM_INDEX_2B
 	const SPRITE_ANIM_INDEX_2C
+
+	const_def
+
+	const SPRITE_ANIM_FRAME_IDX_00
+    const SPRITE_ANIM_FRAME_IDX_01
```
Comment (yenatch) [inline constants/sprite_constants.asm]: please tab indent

**Comment by yenatch** [comment]
Please fix macros/movement.asm.

```
	enum movement_turn_head
turn_head: macro
	db movement_turn_head | \1 ; $00
	endm

__enum__ SET $4
```
```
	enum movement_turn_head
turn_head: macro
	db movement_turn_head + \1
	endm

__enum__ = __enum__ + 3
```

**Pull request merged**